File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/quartz/constant/Concurrent.java
package com.jmsoftware.maf.springcloudstarter.quartz.constant;

import com.jmsoftware.maf.common.enumeration.ValueDescriptionBaseEnum;
import lombok.Getter;
import lombok.ToString;

/**
 * Description: Concurrent, change description here.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com, date: 9/23/2021 2:04 PM
 **/
@Getter
@ToString
public enum Concurrent implements ValueDescriptionBaseEnum<Byte> {
    /**
     * Disallow Concurrent
     */
    DISALLOW_CONCURRENT((byte) 0),
    /**
     * concurrent
     */
    CONCURRENT((byte) 1),
    ;
    private final Byte value;

    Concurrent(Byte value) {
        this.value = value;
    }

    @Override
    public String getDescription() {
        return null;
    }
}

File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/quartz/constant/Concurrent.kt
package com.jmsoftware.maf.springcloudstarter.quartz.constant

/**
 * # Concurrent
 *
 * Description: Concurrent, change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/14/22 10:28 PM
 */
enum class Concurrent(val value: Byte) {
    /**
     * Disallow Concurrent
     */
    DISALLOW_CONCURRENT(0.toByte()),

    /**
     * concurrent
     */
    CONCURRENT(1.toByte());
}
